presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -518,6 +520,22 @@ public Boolean visitParameter(Parameter node, Void context)
             return process(parameters.get(node.getPosition()), context);
         }
 
+        public Boolean visitGroupingOperation(GroupingOperation node, Void context)
+        {
+            List<Expression> argumentsNotInGroupBy = node.getGroupingColumns().stream()
+                    .filter((argument) -> !expressions.contains(argument))
+                    .collect(toImmutableList());
+            if (!argumentsNotInGroupBy.isEmpty()) {
```

Comment:
It would be sufficient to call `findAny` on the stream. Also, the error message could mention the offending expression.